rust/benches/benchmarks/fifo_window.rs

```diff
@@ -0,0 +1,56 @@
+include!("../../tests/common.rs");
+
+use criterion::*;
+// use swag::reactive::Reactive;
+use criterion_cpu_time::PosixTime;
+use swag::recalc::ReCalc;
+use swag::soe::SoE;
+use swag::two_stacks::TwoStacks;
+use swag::FifoWindow;
+
+criterion_group! {
+    name = experiments;
+    config = Criterion::default()
+        .with_measurement(PosixTime::UserTime);
+    targets = suite
+}
+
+fn synthesize(size: u64) -> impl Iterator<Item = Int> {
+    (0..size as i64).map(Int)
+}
+
+fn run<BinOp, Window>(group: &mut BenchmarkGroup<PosixTime>, size_exp: u32, name: &str)
+where
+    Window: FifoWindow<Int, BinOp>,
+    BinOp: Operator,
+{
+    let window = &mut Window::new();
+    let group = group.throughput(Throughput::Elements(1));
+    let size = (2 as u64).pow(size_exp);
+    let data = synthesize(size);
+    // Prefill the window
+    data.for_each(|item| window.push(item));
```

Comment:
I like putting in an assertion here that `window` is in fact the size we expect. Although I just realized that asking the window's size is not defined on any of the traits. I think it would probably make sense on `FifoWindow`.

Reply:
Good point, I will add it! One question, I'm wondering if I should change the `pop` operation to also return an `Option<T>` which contains the value which was popped. WDYT?

Reply:
I think that depends on what is idiomatic for Rust data structures, such as what's in the standard library. In C++, the standard containers that have `pop()` of some form do _not_ return the value, so I believe we implemented them that way in C++. In Rust, if the standard containers that have `pop()` of some form return an `Option<T>`, then I think these should as well. But if their `pop()` also don't return anything, this should follow that.

Reply:
Rust's pops often have `Option<T>` as the return value:
* https://doc.rust-lang.org/std/vec/struct.Vec.html#method.pop
* https://doc.rust-lang.org/std/collections/struct.LinkedList.html#method.pop_back
* https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#method.pop_first

It might not make much sense in windowing but I think it should anyhow be easy to add.